Corona-Warn-App/src/main/java/de/rki/coronawarnapp/submission/server/SubmissionServer.kt

```diff
@@ -0,0 +1,85 @@
+package de.rki.coronawarnapp.submission.server
+
+import com.google.protobuf.ByteString
+import dagger.Lazy
+import de.rki.coronawarnapp.server.protocols.KeyExportFormat
+import de.rki.coronawarnapp.util.PaddingTool.requestPadding
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import timber.log.Timber
+import javax.inject.Inject
+import javax.inject.Singleton
+import kotlin.math.max
+
+@Singleton
+class SubmissionServer @Inject constructor(
+    private val submissionApi: Lazy<SubmissionApiV1>
+) {
+
+    private val api: SubmissionApiV1
+        get() = submissionApi.get()
+
+    data class SubmissionData(
+        val authCode: String,
+        val keyList: List<KeyExportFormat.TemporaryExposureKey>,
+        val consentToFederation: Boolean,
+        val visistedCountries: List<String>
+    )
+
+    suspend fun submitKeysToServer(
+        data: SubmissionData
+    ) = withContext(Dispatchers.IO) {
+        Timber.d("submitKeysToServer()")
+        val authCode = data.authCode
+        val keyList = data.keyList
+        Timber.d("Writing ${keyList.size} Keys to the Submission Payload.")
+
+        val randomAdditions = 0 // prepare for random addition of keys
+        val fakeKeyCount = max(
+            MIN_KEY_COUNT_FOR_SUBMISSION + randomAdditions - keyList.size,
+            0
+        )
+        val fakeKeyPadding = requestPadding(FAKE_KEY_SIZE * fakeKeyCount)
+
+        val submissionPayload = KeyExportFormat.SubmissionPayload.newBuilder()
+            .addAllKeys(keyList)
+            .setPadding(ByteString.copyFromUtf8(fakeKeyPadding))
+            .setConsentToFederation(data.consentToFederation)
+            .addAllVisitedCountries(data.visistedCountries)
+            .build()
+
+        api.submitKeys(
+            authCode = authCode,
+            fake = "0",
+            headerPadding = EMPTY_HEADER,
+            requestBody = submissionPayload
+        )
+    }
+
+    suspend fun submitKeysToServerFake() = withContext(Dispatchers.IO) {
```

Comment:
why this method and not inject fake SubmissionServer when needed?

Reply:
There is lot's of overlap between both methods, and they are both used in the in the playbook, we'd have to inject the real and the fake server, the fake one just uses different parameters.

I'm not opposed to injecting this somehow, just didn't see a clear way to do it when refactoring this. Would leave that for another PR, this one is already pretty huge.